pkg/segment/tracing/handler/tracehandler.go

```diff
@@ -587,3 +587,143 @@ func ProcessDependencyRequest(ctx *fasthttp.RequestCtx, myid uint64) {
+		}
+		log.Errorf("ProcessGanttChartRequest: failed to decode search request body! Err=%v", err)
+	}
+
+	// Parse the JSON data from ctx.PostBody
+	searchRequestBody := &structs.SearchRequestBody{}
+	if err := json.Unmarshal(ctx.PostBody(), &searchRequestBody); err != nil {
+		log.Errorf("ProcessGanttChartRequest: could not unmarshal json body, err=%v", err)
+		return
+	}
+
+	searchRequestBody.QueryLanguage = "Splunk QL"
+	searchRequestBody.IndexName = "traces"
+	searchRequestBody.From = 0
+	searchRequestBody.Size = 1000
+
+	// Used to find out which attributes belong to tags
+	fieldsNotInTag := []string{"trace_id", "span_id", "parent_span_id", "service", "trace_state", "name", "kind", "start_time", "end_time",
+		"duration", "dropped_attributes_count", "dropped_events_count", "dropped_links_count", "status", "events", "links"}
+
+	idToSpanMap := make(map[string]*structs.GanttChartSpan, 0)
+	idToParentId := make(map[string]string, 0)
+
+	for {
+		modifiedData, err := json.Marshal(searchRequestBody)
+		if err != nil {
+			log.Errorf("ProcessGanttChartRequest: could not marshal to json body, err=%v", err)
+			return
+		}
+
+		// Get initial data
+		rawTraceCtx := &fasthttp.RequestCtx{}
+		rawTraceCtx.Request.Header.SetMethod("POST")
+		rawTraceCtx.Request.SetBody(modifiedData)
+		pipesearch.ProcessPipeSearchRequest(rawTraceCtx, myid)
+
+		resultMap := make(map[string]interface{}, 0)
+		err = json.Unmarshal(rawTraceCtx.Response.Body(), &resultMap)
+		if err != nil {
+			log.Errorf("ProcessGanttChartRequest: could not unmarshal json body, err=%v", err)
+			return
+		}
+
+		hits, exists := resultMap["hits"]
+		if !exists {
+			log.Errorf("ProcessGanttChartRequest: Key 'hits' not found in response")
+			break
+		}
+
+		hitsMap, ok := hits.(map[string]interface{})
+		if !ok {
+			log.Errorf("ProcessGanttChartRequest: Error asserting type for 'hits'")
+			break
+		}
+
+		records, exists := hitsMap["records"]
+		if !exists {
+			log.Errorf("ProcessGanttChartRequest: Key 'records' not found in response")
+			break
+		}
+
+		rawSpans, ok := records.([]interface{})
+		if !ok {
+			log.Errorf("ProcessGanttChartRequest: Error asserting type for 'records'")
+			break
+		}
+
+		if len(rawSpans) == 0 {
+			break
+		}
+
+		for _, rawSpan := range rawSpans {
+			spanMap := rawSpan.(map[string]interface{})
+
+			span := &structs.GanttChartSpan{}
+
+			jsonData, err := json.Marshal(spanMap)
+			if err != nil {
+				log.Errorf("ProcessGanttChartRequest: could not marshal to json body, err=%v", err)
+				continue
+			}
+			if err := json.Unmarshal(jsonData, &span); err != nil {
+				log.Errorf("ProcessGanttChartRequest: could not unmarshal to json body, err=%v", err)
+				continue
+			}
+
+			serviceName, exists := spanMap["service"]
+			if !exists {
+				continue
+			}
+
+			operationName, exists := spanMap["name"]
+			if !exists {
+				continue
+			}
+
+			parentSpanId, exists := spanMap["parent_span_id"]
+			if !exists {
+				continue
+			}
```

Comment:
Won't this exclude the root span? I don't think we want to exclude it

Reply:
I don't think I will exclude the root span because the attribute `parent_span_id` exists in the resultMap; it just has a value of `""`. In the subsequent checks, I will determine that if `parent_span_id == ""`, it indicates that it is a root span.